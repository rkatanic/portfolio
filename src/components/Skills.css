.skill-bars {
  margin: 0 0 2rem;
}

.skill-bar {
  position: relative;
  width: 100%;
  height: 4px;
  background-color: var(--color-input);
  z-index: 1;
}

.skill-bar-label {
  margin: 0;
  font-size: 0.825rem;
  margin: 1.5rem 0 0.125rem;
  color: var(--color-title);
}

.skill-bar-percent {
  position: absolute;
  top: 0;
  height: 4px;
  background-color: var(--color-primary);
}

.tool {
  position: relative;
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  justify-content: center;
  flex: 1;
  min-width: 6rem;
  min-height: 7rem;
  padding: 0.5rem;
  background-color: var(--color-input);
  transition: 0.25s ease;
  cursor: pointer;
}

.tool-label {
  font-weight: 600;
  letter-spacing: 2px;
  position: relative;
}

.tool-label::before {
  position: absolute;
  content: "";
  bottom: 0;
  width: 1rem;
  height: 2px;
  background-color: var(--color-primary);
  transition: 0.5s ease;
}

.tool .icon {
  fill: var(--color-primary);
}

.tool:hover {
  transform: scale(1.1);
  background-color: var(--color-container);
}

.tool:hover .tool-label::before {
  width: 2rem;
}

.tools {
  margin: 2rem 0;
  display: grid;
  grid-gap: 0.25rem;
  grid-auto-columns: minmax(0, 1fr);
  grid-auto-flow: column;
  grid-template-areas:
    "a b ."
    "c . d"
    "e f ."
    "g . h";
}

#a {
  grid-area: a;
}

#b {
  grid-area: b;
}

#c {
  grid-area: c;
}

#d {
  grid-area: d;
}

#e {
  grid-area: e;
}

#f {
  grid-area: f;
}

#g {
  grid-area: g;
}

#h {
  grid-area: h;
}
