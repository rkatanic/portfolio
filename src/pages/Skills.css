.txt-date {
  font-family: rustyColaPen;
  text-transform: uppercase;
  letter-spacing: 3px;
}

* {
  cursor: inherit;
}

.div-parent {
  position: relative;
  margin-top: -10vh;
}

.child-txt {
  width: 100%;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
}

.child-shape {
  height: 100vh !important;
}

.heading-sphere {
  top: 0;
  position: absolute;
}

.heading h2 {
  text-align: center;
  line-height: 1;
}

.resume-content {
  margin-bottom: 1.25rem;
}

.resume-content-item {
  margin-bottom: 0.5rem;
  display: flex;
  align-items: baseline;
  justify-content: space-between;
}
.resume-content-item-desc {
  flex: 1;
}
.resume-content-item-desc h2:first-child {
  letter-spacing: 1px;
}

@media screen and (max-width: 768px) {
  .heading {
    text-align: center;
  }

  .heading h2:nth-child(2) {
    text-align: center;
  }
}
