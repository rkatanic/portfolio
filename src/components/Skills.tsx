import Section from "./Section";
import { ReactComponent as WebDevelopmentIcon } from "../assets/icons/web-development.svg";

import "./Skills.css";

const Skills = (): JSX.Element => {
  const skills = [
    { key: "HTML", value: 90 },
    { key: "CSS", value: 90 },
    { key: "JavaScript", value: 80 },
    { key: "Sass", value: 75 },
    { key: "React", value: 75 },
    { key: "Angular", value: 75 },
    { key: "Java", value: 70 },
    { key: "Spring Boot", value: 70 },
    { key: "MySQL", value: 70 },
    { key: "MongoDB", value: 70 },
  ];
  return (
    <Section id="skills">
      <div className="skills">
        <div className="skill-bars">
          <h3>My skills</h3>
          {skills.map(({ key, value }) => (
            <>
              <p className="skill-bar-label">{key}</p>
              <div key={key} className="skill-bar">
                <div
                  className="skill-bar-percent"
                  style={{ width: `${value}%` }}
                ></div>
              </div>
            </>
          ))}
        </div>
        <h3>Services I provide</h3>
        <div className="skills-container">
          <div className="skill-icon">
            <WebDevelopmentIcon />
          </div>
          <h3 className="item-title">Front End Development</h3>
          <p className="item-desc">
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Sequi
            optio quas deleniti incidunt perspiciatis laudantium est odit in
            officiis dolor sit vero veritatis qui aut facilis, adipisci culpa
            exercitationem quam?
          </p>
        </div>
        <div className="skills-container">
          <div className="skill-icon">
            <WebDevelopmentIcon />
          </div>
          <h3 className="item-title">Back End Development</h3>
          <p className="item-desc">
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Sequi
            optio quas deleniti incidunt perspiciatis laudantium est odit in
            officiis dolor sit vero veritatis qui aut facilis, adipisci culpa
            exercitationem quam?
          </p>
        </div>
      </div>
    </Section>
  );
};

export default Skills;
