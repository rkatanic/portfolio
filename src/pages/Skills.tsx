import Section from "../components/Section";
import Services from "../components/Services";
import { ReactComponent as VsCodeIcon } from "../assets/icons/vs-code.svg";
import { ReactComponent as DockerIcon } from "../assets/icons/docker.svg";
import { ReactComponent as GitIcon } from "../assets/icons/git.svg";
import { ReactComponent as GitHubIcon } from "../assets/icons/github.svg";
import { ReactComponent as IntelliJIcon } from "../assets/icons/intellij.svg";
import { ReactComponent as PostmanIcon } from "../assets/icons/postman.svg";
import { ReactComponent as AngularIcon } from "../assets/icons/angular.svg";
import { ReactComponent as ReactIcon } from "../assets/icons/react.svg";
import { ReactComponent as HtmlIcon } from "../assets/icons/html.svg";
import { ReactComponent as CssIcon } from "../assets/icons/css.svg";
import { ReactComponent as SpringBootIcon } from "../assets/icons/spring-boot.svg";
import { ReactComponent as MySqlIcon } from "../assets/icons/mysql.svg";
import { ReactComponent as MongoDbIcon } from "../assets/icons/mongodb.svg";
import { ReactComponent as JavaIcon } from "../assets/icons/java.svg";
import { ReactComponent as JavaScriptIcon } from "../assets/icons/javascript.svg";
import { ReactComponent as SassIcon } from "../assets/icons/sass.svg";

import "./Skills.css";

const Skills = (): JSX.Element => {
  const skills = [
    { name: "HTML", icon: <HtmlIcon /> },
    { name: "CSS", icon: <CssIcon /> },
    { name: "JavaScript", icon: <JavaScriptIcon /> },
    { name: "Java", icon: <JavaIcon /> },
    { name: "Sass", icon: <SassIcon /> },
    { name: "React", icon: <ReactIcon /> },
    { name: "Angular", icon: <AngularIcon /> },
    { name: "Spring Boot", icon: <SpringBootIcon /> },
    { name: "MySQL", icon: <MySqlIcon /> },
    { name: "MongoDB", icon: <MongoDbIcon /> },
    { name: "Postman", icon: <PostmanIcon /> },
    { name: "Docker", icon: <DockerIcon /> },
    { name: "Git", icon: <GitIcon /> },
    { name: "GitHub", icon: <GitHubIcon /> },
    { name: "VS Code", icon: <VsCodeIcon /> },
    { name: "IntelliJ", icon: <IntelliJIcon /> },
  ];

  return (
    <Section id="skills">
      <div className="skills">
        <h3 className="headline">
          Languages <span className="primary">&</span> tools i use
        </h3>
        <div className="skill-list">
          {skills.map(({ name, icon }) => (
            <div key={name} className="skill-card">
              <div className="highlight"></div>
              {icon}
              <p className="skill-card-label">{name}</p>
            </div>
          ))}
        </div>
        <Services />
      </div>
    </Section>
  );
};

export default Skills;
