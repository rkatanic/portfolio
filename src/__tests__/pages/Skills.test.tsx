import { render } from "@testing-library/react";
import Skills from "../../pages/Skills";

describe("Skills", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Skills />);

    expect(baseElement).toMatchSnapshot();
  });
});
