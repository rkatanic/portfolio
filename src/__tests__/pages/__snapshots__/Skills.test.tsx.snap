// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should render 1`] = `
<body>
  <div>
    <section
      class="section"
      id="skills"
    >
      <div
        class="skills"
      >
        <h3
          class="headline"
        >
          Languages 
          <span
            class="primary"
          >
            &
          </span>
           tools i use
        </h3>
        <div
          class="skill-list"
        >
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              html.svg
            </svg>
            <p
              class="skill-card-label"
            >
              HTML
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              css.svg
            </svg>
            <p
              class="skill-card-label"
            >
              CSS
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              javascript.svg
            </svg>
            <p
              class="skill-card-label"
            >
              JavaScript
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              java.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Java
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              sass.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Sass
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              react.svg
            </svg>
            <p
              class="skill-card-label"
            >
              React
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              angular.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Angular
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              spring-boot.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Spring Boot
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              mysql.svg
            </svg>
            <p
              class="skill-card-label"
            >
              MySQL
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              mongodb.svg
            </svg>
            <p
              class="skill-card-label"
            >
              MongoDB
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              postman.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Postman
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              docker.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Docker
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              git.svg
            </svg>
            <p
              class="skill-card-label"
            >
              Git
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              github.svg
            </svg>
            <p
              class="skill-card-label"
            >
              GitHub
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              vs-code.svg
            </svg>
            <p
              class="skill-card-label"
            >
              VS Code
            </p>
          </div>
          <div
            class="skill-card"
          >
            <div
              class="highlight"
            />
            <svg>
              intellij.svg
            </svg>
            <p
              class="skill-card-label"
            >
              IntelliJ
            </p>
          </div>
        </div>
        <h3
          class="headline"
        >
          Services I provide
        </h3>
        <div
          class="service-cards"
        >
          <div
            class="service-card"
          >
            <div
              class="service-card-content"
            >
              <h3
                class="card-title"
              >
                Front-End Development
              </h3>
              <p
                class="service-card-content-desc"
              >
                Using modern languages and frameworks, I can develop fast and responsive single-page client applications.
              </p>
              <div
                class="service-card-content-badges"
              >
                <b
                  class="service-card-content-badge"
                >
                  HTML
                </b>
                <b
                  class="service-card-content-badge"
                >
                  CSS
                </b>
                <b
                  class="service-card-content-badge"
                >
                  JavaScript
                </b>
                <b
                  class="service-card-content-badge"
                >
                  Sass
                </b>
                <b
                  class="service-card-content-badge"
                >
                  React
                </b>
                <b
                  class="service-card-content-badge"
                >
                  Angular
                </b>
              </div>
            </div>
          </div>
          <div
            class="service-card"
          >
            <div
              class="service-card-content"
            >
              <h3
                class="card-title"
              >
                Back-End Development
              </h3>
              <p
                class="service-card-content-desc"
              >
                I can develop scalable and maintainable server applications in Java, using Spring Boot framework.
              </p>
              <div
                class="service-card-content-badges"
              >
                <b
                  class="service-card-content-badge"
                >
                  Java
                </b>
                <b
                  class="service-card-content-badge"
                >
                  Spring Boot
                </b>
                <b
                  class="service-card-content-badge"
                >
                  MySQL
                </b>
                <b
                  class="service-card-content-badge"
                >
                  MongoDB
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
